pkg/controller/core/clusterqueue_controller.go

```diff
@@ -93,6 +96,10 @@ func (r *ClusterQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		if !r.cache.ClusterQueueTerminating(cqObj.Name) {
 			r.cache.TerminateClusterQueue(cqObj.Name)
 		}
+		err := r.UpdateCQStatusIfChanged(ctx, &cqObj, metav1.ConditionFalse, "Terminating", cqTerminatingStatusMsg)
```

Comment:
We don't need the logic here.

If we are able to remove the finalizer below, then this object will be deleted. Then adding the condition is just a waste.

Reply:
Makes sense.

Reply:
Done.